source/codegen/templates/service_helpers.mako

```diff
@@ -294,12 +294,14 @@ one_of_case_prefix = f'{namespace_prefix}{function_name}Request::{PascalFieldNam
         if(${iterator_name} != ${map_name}.end()) {
           response->set_${parameter_name}(static_cast<${namespace_prefix}${parameter["enum"]}>(${iterator_name}->second));
         }
+%     elif common_helpers.is_array(parameter['type']) and common_helpers.is_string_arg(parameter):
+        CopyEnumValues(${parameter_name}.data(), response->mutable_${parameter_name}(), ${parameter_name}.size());
```

Comment:
You need to add similar `CopyEnumValues` statement in `if` block above (lines 287 to 296) and pass the map as last parameter. Above block seems to handle only scalar enum parameters. You can add this condition for it to handle array enum output parameter as well. You can validate codegen for that by using `FloatEnum` for the nifake API you are using for tests above and probably changing data type of the enum parameter to `float`. [FloatEnum](https://github.com/doshirohan/grpc-device/blob/main/source/codegen/metadata/nifake/enums.py#L96) has `generate-mappings` set to true. So using that for function parameter would trigger above block of codegen.

Reply:
added